packages/vite/client.d.ts

```diff
@@ -164,15 +164,17 @@ declare module '*.webmanifest' {
 
 // web worker
 declare module '*?worker' {
+  import { Worker as NodeWorker } from 'worker_threads'
   const workerConstructor: {
-    new (): Worker
+    new (): Worker | NodeWorker
```

Comment:
It could protentially break the existing code, and also make it relying on `@types/node`. Given this could be a fairly rare use case, I would think it's better to be done on the client side.

Reply:
Good catch

Reply:
Yeah existing TS code may break. Will update PR :)

Reply:
@antfu Updated PR